docs/API.md

```diff
@@ -201,6 +201,19 @@ Go back one entry in the history.
 ##### `goForward()`
 Go forward one entry in the history.
 
+##### `setRouteLeaveHook(route, hook)`
+Registers the given hook function to run before leaving the given route.
+
+During a normal transition, the hook function receives the next location
+as its only argument. Returning false will abort the transition. Any other return values will not have any effect.
```

Comment:
> During a normal transition, the hook function receives the next location as its only argument and can return a prompt message (string) to show the user, to make sure they want to leave the page, false, to prevent the transition. Any other return values will not have any effect.